tests/components/binary_sensor/test_rflink.py

```diff
@@ -0,0 +1,161 @@
+    assert config_sensor.attributes['device_class'] == 'door'
+
+    # test event for config sensor
+    event_callback({
+        'id': 'test',
+        'command': 'on',
+    })
+    yield from hass.async_block_till_done()
+
+    assert hass.states.get('binary_sensor.test').state == STATE_ON
+
+    # test event for config sensor
+    event_callback({
+        'id': 'test',
+        'command': 'off',
+    })
+    yield from hass.async_block_till_done()
+
+    assert hass.states.get('binary_sensor.test').state == STATE_OFF
+
+
+@asyncio.coroutine
+def test_entity_availability(hass, monkeypatch):
+    """If Rflink device is disconnected, entities should become unavailable."""
+    # Make sure Rflink mock does not 'recover' to quickly from the
+    # disconnect or else the unavailability cannot be measured
+    config = CONFIG
+    failures = [True, True]
+    config[CONF_RECONNECT_INTERVAL] = 60
+
+    # Create platform and entities
+    event_callback, create, _, disconnect_callback = yield from mock_rflink(
+        hass, config, DOMAIN, monkeypatch, failures=failures)
+
+    # Entities are available by default
+    assert hass.states.get('binary_sensor.test').state == STATE_OFF
+
+    # Mock a disconnect of the Rflink device
+    disconnect_callback()
+
+    # Wait for dispatch events to propagate
+    yield from hass.async_block_till_done()
+
+    # Entity should be unavailable
+    assert hass.states.get('binary_sensor.test').state == STATE_UNAVAILABLE
+
+    # Reconnect the Rflink device
+    disconnect_callback()
+
+    # Wait for dispatch events to propagate
+    yield from hass.async_block_till_done()
+
+    # Entities should be available again
+    assert hass.states.get('binary_sensor.test').state == STATE_OFF
+
+
+@asyncio.coroutine
+def test_off_delay(hass, monkeypatch):
+    """Test off_delay option."""
+    # setup mocking rflink module
+    event_callback, create, _, disconnect_callback = yield from mock_rflink(
+        hass, CONFIG, DOMAIN, monkeypatch)
+
+    # make sure arguments are passed
+    assert create.call_args_list[0][1]['ignore']
+
+    events = []
+
+    @ha.callback
+    def callback(event):
+        """Verify event got called."""
+        events.append(event)
+
+    hass.bus.async_listen(EVENT_STATE_CHANGED, callback)
+
+    # turn on sensor
+    event_callback({
+        'id': 'test2',
+        'command': 'on',
+    })
+    yield from hass.async_block_till_done()
+    state = hass.states.get('binary_sensor.test2')
+    assert state.state == STATE_ON
+    assert len(events) == 1
+
+    # turn on sensor again
+    event_callback({
+        'id': 'test2',
+        'command': 'on',
+    })
+    yield from hass.async_block_till_done()
+    state = hass.states.get('binary_sensor.test2')
+    assert state.state == STATE_ON
+    assert len(events) == 2
+
+    # fake time change and verify sensor sent a single off event
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=30))
```

Comment:
Probably patch `dt_util.utcnow()` from the beginning of setup, to make sure we don't get flaky tests. Make sure to patch in the correct place, ie where it is used within the code under test.

Reply:
Thanks!